src/com/softwaret/kpdf/response/error/factory/ErrorFactoryImpl.kt

```diff
@@ -0,0 +1,14 @@
+package com.softwaret.kpdf.response.error.factory
+
+import ErrorFactory
+import com.softwaret.kpdf.response.error.Error
+import com.softwaret.kpdf.response.error.ErrorType
+
+class ErrorFactoryImpl : ErrorFactory {
+
+   override fun build(errorType: ErrorType) =
+        when (errorType) {
+            ErrorType.NULL_EMPTY_LOGIN -> Error(errorType.name, errorType.message)
```

Comment:
Po co nam to rozbicie na dwa błędy? Jaki jest w sumie cel tej fabryki? Każdy błąd mógłby sam umieć się budować jeżeli użyjemy sealed class z bazową metdoą

Reply:
Jeżeli tak zrobimy to będziemy mogli wywalić cały flow w którym teraz używamy `Error`. Sama nazwa `Error` nie mówi do czego służy ta klasa i czym różni się od Exception albo Throwable

Reply:
Fabryka jest do wywalenia